File: app/src/main/java/theredspy15/ltecleanerfoss/CleanReceiver.java
package theredspy15.ltecleanerfoss;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.SystemClock;

public class CleanReceiver extends BroadcastReceiver {
    private static final int PERIOD = 86400000;
    private static final int INITIAL_DELAY = 3600000; // 60 seconds

    @Override
    public void onReceive(Context ctxt, Intent i) {
        if (i.getAction() == null) {
            ScheduledService.enqueueWork(ctxt);
        } else {
            scheduleAlarm(ctxt);
        }
    }

    public static void scheduleAlarm(Context ctxt) {
        AlarmManager mgr =
                (AlarmManager) ctxt.getSystemService(Context.ALARM_SERVICE);
        Intent i = new Intent(ctxt, CleanReceiver.class);
        PendingIntent pi;
        pi = PendingIntent.getBroadcast(ctxt, 0, i, PendingIntent.FLAG_UPDATE_CURRENT);

        mgr.setRepeating(AlarmManager.ELAPSED_REALTIME,
                SystemClock.elapsedRealtime() + INITIAL_DELAY,
                PERIOD, pi);
    }

    public static void cancelAlarm(Context ctxt) {
        AlarmManager mgr =
                (AlarmManager) ctxt.getSystemService(Context.ALARM_SERVICE);
        Intent i = new Intent(ctxt, CleanReceiver.class);
        PendingIntent pi;
        pi = PendingIntent.getBroadcast(ctxt, 0, i, PendingIntent.FLAG_CANCEL_CURRENT);

        mgr.cancel(pi);
    }
}

File: app/src/main/java/theredspy15/ltecleanerfoss/CleanReceiver.kt
package theredspy15.ltecleanerfoss

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.SystemClock
import theredspy15.ltecleanerfoss.ScheduledService.Companion.enqueueWork

class CleanReceiver : BroadcastReceiver() {
    override fun onReceive(ctxt: Context, i: Intent) {
        if (i.action == null) {
            enqueueWork(ctxt)
        } else {
            scheduleAlarm(ctxt)
        }
    }

    companion object {
        private const val PERIOD = 86400000
        private const val INITIAL_DELAY = 3600000 // 60 seconds
        @JvmStatic
        fun scheduleAlarm(context: Context) {
            val mgr = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
            val i = Intent(context, CleanReceiver::class.java)
            val pi: PendingIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
                PendingIntent.getBroadcast(context, 0, i, PendingIntent.FLAG_IMMUTABLE)
            } else {
                PendingIntent.getBroadcast(context, 0, i, PendingIntent.FLAG_UPDATE_CURRENT)
            }
            mgr.setRepeating(
                AlarmManager.ELAPSED_REALTIME,
                SystemClock.elapsedRealtime() + INITIAL_DELAY,
                PERIOD.toLong(), pi
            )
        }

        @JvmStatic
        fun cancelAlarm(context: Context) {
            val mgr = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
            val i = Intent(context, CleanReceiver::class.java)
            val pi: PendingIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
                PendingIntent.getBroadcast(context, 0, i, PendingIntent.FLAG_IMMUTABLE)
            } else {
                PendingIntent.getBroadcast(context, 0, i, PendingIntent.FLAG_CANCEL_CURRENT)
            }
            mgr.cancel(pi)
        }
    }
}